apis/Google.Devtools.AspNet/Google.Devtools.AspNet/CloudErrorReporting/CloudErrorReportingExceptionLogger.cs

```diff
@@ -94,15 +100,32 @@ public sealed class CloudErrorReportingExceptionLogger : ExceptionLogger
         /// <inheritdoc />
         public override Task LogAsync(ExceptionLoggerContext context, CancellationToken cancellationToken)
         {
-            ReportedErrorEvent errorEvent = CreateReportRequest(context);
-            return _client.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            if (_clientTask.IsCompleted)
+            {
+                ReportedErrorEvent errorEvent = CreateReportRequest(context);
+                return _clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
+            }
+
+            if (Interlocked.Increment(ref _taskCounter) <= MaxWaitingTasks)
+            {
+                return _clientTask.ContinueWith(clientTask =>
+                {
+                    if (clientTask.IsFaulted)
+                    {
+                        throw new InvalidOperationException("The client task has faulted.");
+                    }
+                    ReportedErrorEvent errorEvent = CreateReportRequest(context);
+                    return clientTask.Result.ReportErrorEventAsync(_projectResourceName, errorEvent);
+                });
+            }
+
+            return Task.FromResult(false);
         }
 
         /// <inheritdoc />
         public override void Log(ExceptionLoggerContext context)
         {
-            ReportedErrorEvent errorEvent = CreateReportRequest(context);
-            _client.ReportErrorEvent(_projectResourceName, errorEvent);
+            LogAsync(context, CancellationToken.None).Wait();
```

Comment:
I mean that `Log` should basically be:

```csharp
var client = _client.Result;
client.Log(context);
```

In general, I try to avoid wrapping sync-in-async or vice versa unless I have to.